### PR TITLE
Official example: try moving options to `manager.js`

### PR DESCRIPTION
**Pull request by Hypnosphi**
I tried applying https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#using-managerjs to our own example storybook

I found two issues:
1. `showRoots` doesn't work
2. `manager.js` doesn't hot reload: e.g. if you change it to `theme: themes.dark`, it applies only after page reload

### CODE REVIEW

**Comment by ndelangen** [comment]
For 6.0.0 the documentation is correct.

For 5.3 we need to reverse this documentation a bit, to tell users to keep setting the global parameter

**Comment by ndelangen** [comment]
I can confirm that:
> showRoots doesn't work

I added a FIX in `api/modules/stories.ts`

**Comment by ndelangen** [comment]
@tmeasday I did find what @Hypnosphi found too:

When a user would make migrate all options to `manager.js` certain things break.

- storySort (I added a fix for this, in this PR)
- showRoots (I added a fix for this, in this PR)
- theme *(the preview never gets the theme, meaning addon-docs isn't themed the same)* (I opened a new ticket for this: https://github.com/storybookjs/storybook/issues/9662)

@Hypnosphi WDYT?

after this PR, this becomes perfectly valid config:
```js
// manager.js
import { addons } from '@storybook/addons';
import { themes } from '@storybook/theming';

addons.setConfig({
  showRoots: true,
  theme: themes.light, // { base: 'dark', brandTitle: 'Storybook!' },
  storySort: (a, b) =>
    a[1].kind === b[1].kind ? 0 : a[1].id.localeCompare(b[1].id, undefined, { numeric: true }),
});
```

**Comment by ndelangen** [comment]
`manager.js` not hot-module-reloading is fine by me for now. I don not consider this an critical important feature. I'd be nice to add in the future.

File: lib/api/src/modules/stories.ts [hunk 1]
```diff
@@ -229,7 +230,7 @@ const initStoriesApi = ({
         hierarchyRootSeparator: rootSeparator = undefined,
         hierarchySeparator: groupSeparator = undefined,
         showRoots = undefined,
-      } = (parameters && parameters.options) || {};
```
Comment (tmeasday) [inline lib/api/src/modules/stories.ts]: @ndelangen do you think we should come up with a principled way to pull configuration from both parameters and `.getConfig()`? It seems like this comes up a bit and maybe there should be a single API for it?

File: lib/api/src/modules/stories.ts [hunk 2]
```diff
@@ -223,94 +223,99 @@ const initStoriesApi = ({
...
-
-        // 3. If the user passes showRoots, or doesn't match above, do a simpler splitting.
-      } else {
-        const parts: string[] = kind.split('/');
-        if (showRoots && parts.length > 1) {
-          [root, ...groups] = parts;
+    const { storySort = () => 0 } = provider.getConfig();
+
+    const storiesHashOutOfOrder = Object.entries(input)
+      .sort(storySort)
```
Comment (tmeasday) [inline lib/api/src/modules/stories.ts]: Am I right in saying that if you set the `storySort` options parameter it sorts the story on the preview side (and so they'll be sorted e.g. in `store.extract()`), but if you set the addon option it sorts them only in the UI here?

That seem... super confusing, and likely to cause problems (if for Chromatic + other integrators if nothing else).

I would vote you just **CANNOT** set `storySort` in `manager.js`. It is a property of the store, not a UI option. With that in mind it probably shouldn't live in `parameters.options`. 🤷‍♂

**Review by tmeasday — CHANGES_REQUESTED** [review]

File: lib/api/src/modules/stories.ts (hunk 1 above)
Comment (tmeasday) [reply lib/api/src/modules/stories.ts]: For instance @atanasster does something very similar in https://github.com/storybookjs/storybook/pull/9095

File: lib/api/src/modules/stories.ts (hunk 2 above)
Comment (ndelangen) [reply lib/api/src/modules/stories.ts]: I'm considering putting it in `main.js` and making it a hard requirement it's a pure function.

That way I can inject the function into both environments...

Comment (tmeasday) [reply lib/api/src/modules/stories.ts]: I'm not necessarily against putting it in `main.js` but why do we need it in both environments?

Comment (ndelangen) [reply lib/api/src/modules/stories.ts]: I'm looking into this today.

File: lib/api/src/modules/stories.ts [hunk 3]
```diff
@@ -222,94 +223,96 @@ const initStoriesApi = ({
...
+
+        // Kind splitting behaviour as per https://github.com/storybookjs/storybook/issues/8793
+        let root = '';
+        let groups: string[];
+        // 1. If the user has passed separators, use the old behaviour but warn them
+        if (typeof rootSeparator !== 'undefined' || typeof groupSeparator !== 'undefined') {
+          warnRemovingHierarchySeparators();
+          if (usingShowRoots) warnUsingHierarchySeparatorsAndShowRoots();
+          ({ root, groups } = parseKind(kind, {
+            rootSeparator: rootSeparator || '|',
```
Comment (shilman) [inline lib/api/src/modules/stories.ts]: Are we removing these in 6.0? Or are we planning to patch this back to 5.3? Or will we do that in a separate PR?

Comment (ndelangen) [reply lib/api/src/modules/stories.ts]: Yeah removing those deprecations is on my wishlist

Comment (shilman) [reply lib/api/src/modules/stories.ts]: I know a single key on your keyboard that could make your wish come true 😉

Comment (ndelangen) [reply lib/api/src/modules/stories.ts]: I'm well aware of existence of the `delete` key on my keyboard

Comment (ndelangen) [reply lib/api/src/modules/stories.ts]: Screenshot 2020-02-13 at 14 59 27

**Review by ndelangen — APPROVED** [review]

**Comment by tmeasday** [comment]
👏
